### PR TITLE
ISPN-8989 Administration console - Counters don't work in standalone mode

### PR DESCRIPTION
**Pull request by vblagoje**
@ryanemerson we can merge this one into 9.2 and master, of course.

### CODE REVIEW

File: src/module/cache-container/view/cache-container.html [hunk 1]
```diff
@@ -75,7 +75,7 @@ <h1><span class="label"
     <div class="col-sm-12 col-md-12">
       <ul class="nav nav-tabs">
         <li ui-sref-active="active" ui-sref=".caches"><a translate>CACHES</a></li>
-        <li ui-sref-active="active" ui-sref=".counters" ng-if="!ctrl.isLocalMode()"><a translate>COUNTERS</a></li>
+        <li ui-sref-active="active" ui-sref=".counters"><a translate>COUNTERS</a></li>
```
Comment (ryanemerson) [inline src/module/cache-container/view/cache-container.html]: This is strange, even though this looks correct when I load up the server with `standalone.sh` no counters tab is present.

File: src/services/counters/CounterService.ts [hunk 2]
```diff
@@ -102,12 +114,19 @@ export class CounterService {
...
   reset(container: ICacheContainer, counter: ICounter): ng.IPromise<any> {
-    return this.jGroupsService.getServerGroupCoordinator(container.serverGroup).then((coord: IServerAddress) => {
+    if (this.launchType.isStandaloneLocalMode()) {
       return this.dmrService.executePost({
-        address: this.getCounterRuntimeAddress(counter, container.name, coord.host, coord.name),
+        address: this.getCounterRuntimeAddress(counter, container.name),
         operation: "counter-reset"
       });
-    });
+    } else {
```
Comment (ryanemerson) [inline src/services/counters/CounterService.ts]: You don't handle `isStandaloneClusteredMode` here which I think is why it fails with standalone.sh -c clustered.xml

File: src/services/counters/CounterService.ts [hunk 3]
```diff
@@ -52,42 +52,54 @@ export class CounterService {
...
-            }
-          }, trail);
-          return counters;
-        }).then(counters => {
+      this.findCounters(request).then(counters => {
         deferred.resolve(counters);
       });
-    });
+    } else {
+      this.jGroupsService.getServerGroupCoordinator(cacheContainer.serverGroup).then((coord: IServerAddress) => {
```
Comment (ryanemerson) [inline src/services/counters/CounterService.ts]: No need for typing here, can just be `coord => {`

File: src/services/counters/CounterService.ts [hunk 4]
```diff
@@ -52,42 +52,54 @@ export class CounterService {
...
+        this.findCounters(request).then(counters => {
+          deferred.resolve(counters);
+        });
+      });
+    }
     return deferred.promise;
   }
 
+  findCounters(request: IDmrRequest): ng.IPromise<ICounter[]> {
+    return this.dmrService.readResource(request).then((response: any): ICounter[] => {
```
Comment (ryanemerson) [inline src/services/counters/CounterService.ts]: No need for typing on lambda

File: src/module/cache-container/view/cache-container.html (hunk 1 above)
Comment (vblagoje) [reply src/module/cache-container/view/cache-container.html]: Try using incognito windows...I'll try again myself.

File: src/services/counters/CounterService.ts (hunk 2 above)
Comment (vblagoje) [reply src/services/counters/CounterService.ts]: I'll check - it works for me. Don't forget that you also need to remove counters=COUNTERS on fresh restart of the test between standalone and clustered

File: src/services/counters/CounterService.ts (hunk 3 above)
Comment (vblagoje) [reply src/services/counters/CounterService.ts]: Ok, I see what you mean. It is easier for me to think about the object, functions, and code in general if it's type is declared.

File: src/services/counters/CounterService.ts (hunk 4 above)
Comment (vblagoje) [reply src/services/counters/CounterService.ts]: ok

**Comment by vblagoje** [comment]
Ok @ryanemerson I'll recheck everything once again. Some errors might have slipped by.

File: src/services/counters/CounterService.ts (hunk 2 above)
Comment (ryanemerson) [reply src/services/counters/CounterService.ts]: Why? If you are starting the server with different configurations then the state is persisted to the currently loaded config. So starting the server with `standalone.sh` and `standalone.sh -c clustered.xml` should be separate.

**Comment by ryanemerson** [comment]
@vblagoje Note I am running the serve with `gulp serve:dist` which is how the console behaves once distributed, so that might explain why we're experiencing different things. 
- `standalone.sh` counters tab is still not present
- `standalone.sh -c clustered.xml` now works :+1:

**Comment by ryanemerson** [comment]
@vblagoje Ignore my last comment, running `gulp clean` explicitly solved the problem. Thanks

**Pull request merged**